tests/cobbler_collections/system_collection_test.py

```diff
@@ -0,0 +1,712 @@
+from typing import Callable
+
+import pytest
+
+from cobbler.api import CobblerAPI
+from cobbler.cexceptions import CX
+from cobbler.cobbler_collections import systems
+from cobbler.cobbler_collections.manager import CollectionManager
+from cobbler.items import distro, profile, system
+
+
+@pytest.fixture
+def system_collection(collection_mgr: CollectionManager):
+    """
+    Fixture to provide a concrete implementation (Systems) of a generic collection.
+    """
+    return systems.Systems(collection_mgr)
+
+
+def test_obj_create(collection_mgr: CollectionManager):
+    # Arrange & Act
+    system_collection = systems.Systems(collection_mgr)
+
+    # Assert
+    assert isinstance(system_collection, systems.Systems)
+
+
+def test_factory_produce(cobbler_api: CobblerAPI, system_collection: systems.Systems):
+    # Arrange & Act
+    result_system = system_collection.factory_produce(cobbler_api, {})
+
+    # Assert
+    assert isinstance(result_system, system.System)
+
+
+def test_get(
+    cobbler_api: CobblerAPI,
+    create_distro: Callable[[str], distro.Distro],
+    create_profile: Callable[[str], profile.Profile],
+    create_system: Callable[[str], system.System],
+):
+    # Arrange
+    name = "test_get"
+    distro1 = create_distro()
+    profile1 = create_profile(name)
+    system1 = create_system(name)
+    system_collection = cobbler_api.systems()
+
+    # Act
+    assert isinstance(system1, system.System)
```

Comment:
Why is this check in the Act section? Should it be moved to Assert or even cut, since it is testing create_system?

Reply:
That check is duplicate indeed. Removing it.

Reply:
I will remove the check.